eng/template/README.md.tpl

```diff
@@ -0,0 +1,76 @@
+# Azure {{PackageTitle}} Module for Go
+
+[![PkgGoDev](https://pkg.go.dev/badge/github.com/Azure/azure-sdk-for-go/sdk/{{rpName}}/{{packageName}})](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/{{rpName}}/{{packageName}})
```

Comment:
Based on our other conversations this will always end up being a broken link for new libraries. Have we came up with a strategy for that?

Reply:
Unfortunately no. We do not really want to ship this without the link - this will be the link to package references

Reply:
We at least need to pick a strategy for folks because this will break in CI for every new library we use it on.

Reply:
I once made a change to the ci.yml template to make sure we could go over every step in the build analysis job by moving the link verification to the last step). In this way, when we are onboarding a new RP, we can ensure that if the only failure in this PR is the link verification, and the RP is generated from a specific template, this PR should be fine and the invalid link will be good after the PR is merged and the tag is added. This is only current practical way to do service onboarding.

Reply:
I suppose that is one option but it does mean that you will have to override any PR with changes to that file before it is release the first time. It isn't ideal and if we get the link wrong or things change/rename before we release we risk shipping a broken link here. 

I personally like leaving this commented out by default and after our first release we can uncomment it.